### PR TITLE
Add default weight rule configuration option

### LINKED ISSUE #12807: add default weight_rule to airflow.cfg
**Description**

It would be nice if the weight_rule default value could be managed by a global config

suggested config:
```
# Weighting method used for the effective total priority weight of the task. 
# Options are: { downstream | upstream | absolute } default is
default_weight_rule = downstream
```
**Use case / motivation**

In some pipeline, you really need to have absolute weight, and then you have to add a line in each task definition which is annoying

### PR DESCRIPTION
**Pull request by eladkal**
closes: https://github.com/apache/airflow/issues/12807

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -497,7 +497,7 @@ def __init__(
         params: Optional[Dict] = None,
         default_args: Optional[Dict] = None,
         priority_weight: int = 1,
-        weight_rule: str = WeightRule.DOWNSTREAM,
+        weight_rule: str = conf.get('core', 'default_task_weight_rule'),
```
Comment (ephraimbuddy) [inline airflow/models/baseoperator.py]:
```suggestion
        weight_rule: str = conf.get('core', 'default_task_weight_rule', fallback=WeightRule.DOWNSTREAM),
```
Should we have a fallback too?

Comment (eladkal) [reply airflow/models/baseoperator.py]: added

File: airflow/config_templates/default_airflow.cfg [hunk 2]
```diff
@@ -207,6 +207,9 @@ dag_discovery_safe_mode = True
 # The number of retries each task is going to have by default. Can be overridden at dag or task level.
 default_task_retries = 0
 
+# The weighting method used for the effective total priority weight of the task
+default_task_weight_rule = downstream
```
Comment (ashb) [inline airflow/config_templates/default_airflow.cfg:211]: We should probably validate that this option is valid (i.e. it's in the enum) in airflow.configuration.Configration.validate()

Comment (eladkal) [reply airflow/config_templates/default_airflow.cfg:211]: is https://github.com/apache/airflow/pull/18627/commits/da521bd820ead09cfd6fa1d1b8f1c73f16a66c78 what you had in mind?

File: airflow/configuration.py [hunk 3]
```diff
@@ -193,6 +194,8 @@ class AirflowConfigParser(ConfigParser):
         },
     }
 
+    enums = {"default_task_weight_rule": WeightRule}
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
    enums = {("core", "default_task_weight_rule"): WeightRule}
```

File: airflow/configuration.py [hunk 4]
```diff
@@ -229,6 +234,14 @@ def validate(self):
 
         self.is_validated = True
 
+    def _validate_enums(self):
+        """Validate that enum type config has an accepted value"""
+        for setting, enum_class in self.enums.items():
+            if self.has_option('core', setting):
+                value = self.get("core", setting)
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
        for (section, setting), enum_class in self.enums.items():
            if self.has_option(section, setting):
                value = self.get(section, setting)
```

File: airflow/configuration.py [hunk 5]
```diff
@@ -229,6 +234,14 @@ def validate(self):
...
+                if not enum_class.is_valid(value):
+                    raise AirflowConfigException(f"{value} is not an accepted config for {setting}")
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
                    raise AirflowConfigException(f"{value} is not an accepted config for  [{section}] {setting}")
```

File: airflow/config_templates/default_airflow.cfg (hunk 2 above)
Comment (ashb) [reply airflow/config_templates/default_airflow.cfg:211]: Yup, thanks!

Comment (ashb) [reply airflow/config_templates/default_airflow.cfg:211]: Oh. Though it turns out WeightRule isn't actually an Enum. Go figure.

File: airflow/configuration.py [hunk 6]
```diff
@@ -229,6 +234,14 @@ def validate(self):
 
         self.is_validated = True
 
+    def _validate_enums(self):
+        """Validate that enum type config has an accepted value"""
+        for setting, enum_class in self.enums.items():
+            if self.has_option('core', setting):
+                value = self.get("core", setting)
+                if not enum_class.is_valid(value):
```
Comment (ashb) [inline airflow/configuration.py:242]: For an actaul enum (i.e. one inheriting from `enum.Enum`) the test for this is

```
try:
    enum_class(value)
except ValueError:
    raise AirflowConfigException(f"{value} is not an accepted config for {setting}") from None
```

File: airflow/config_templates/default_airflow.cfg (hunk 2 above)
Comment (eladkal) [reply airflow/config_templates/default_airflow.cfg:211]: It was supposed to be converted to Enum in https://github.com/apache/airflow/pull/5302

File: airflow/configuration.py (hunk 6 above)
Comment (eladkal) [reply airflow/configuration.py:242]: yeah that should be done after https://github.com/apache/airflow/pull/5302

**Review by ashb — APPROVED** [review]

**Pull request merged**